Source/Types/Bootstrap/Boot.cs

```diff
@@ -24,12 +25,32 @@ public class Boot
         /// <returns><see cref="ITypeFinder"/> that can be used</returns>
         public static ITypeFinder Start(IAssemblies assemblies, IScheduler scheduler, ILogger logger, Assembly entryAssembly = null)
         {
-            var contractToImplementorsMap = new ContractToImplementorsMap(scheduler);
+            if (entryAssembly == null) entryAssembly = Assembly.GetEntryAssembly();
 
-            if( entryAssembly == null ) entryAssembly = Assembly.GetEntryAssembly();
-            contractToImplementorsMap.Feed(entryAssembly.GetTypes());
-            
-            var typeFinder = new TypeFinder(assemblies, contractToImplementorsMap, scheduler, logger);
+            IContractToImplementorsMap contractToImplementorsMap;
+            // Re-enable when https://github.com/dolittle-fundamentals/DotNET.Fundamentals/issues/219 is fixed
+#if(false)
+            if (CachedContractToImplementorsMap.HasCachedMap(entryAssembly))
+            {
+                var before = DateTime.UtcNow;
+                logger.Information("Contract to implementors map cache found - using it instead of dynamically discovery");
+                contractToImplementorsMap = new CachedContractToImplementorsMap(
+                    new ContractToImplementorsSerializer(logger), 
+                    entryAssembly);
+                Console.WriteLine($"CachedMap : {DateTime.UtcNow.Subtract(before).ToString("G")}");
```

Comment:
Use ILogger instead?

Reply:
Due to logging taking a significant amount of CPU, I needed to measure timing without logging on to get more accurate data. Since this is WIP - we can keep it. It is not used right now anyways.